bin/lib/resolve_git_commit_sha_from_image.py

```diff
@@ -0,0 +1,115 @@
+#! /usr/bin/env python3
+
+'''
+Contains functions for validating and resolving image tags.
+'''
+
+import argparse
+import json
+import re
+import urllib.error
+import urllib.request
+import sys
+
+_CIVIFORM_RELEASE_TAG_REGEX = re.compile(r'^v[0-9]+\.[0-9]+\.[0-9]+$')
+
+_DOCKERHUB_REGISTRY_URL = 'https://registry-1.docker.io/v2'
+
+
+class _TagOrDigestDoesNotExistError(Exception):
+
+    def __init__(self, tag_or_digest):
+        super().__init__(f'"{tag_or_digest}" could not be found in DockerHub.')
+
+
+class _NoGitCommitInfoError(Exception):
+
+    def __init__(self, tag_or_digest):
+        super().__init__(
+            f'Git commit information could not be obtained for "{tag_or_digest}"')
+
+
+def _maybe_warn_for_unsupported_tag(tag_or_digest):
+    if _CIVIFORM_RELEASE_TAG_REGEX.match(tag_or_digest):
+        return True
+
+    # Warn the user that this tag type may not be stable and prompt them
+    # to continue.
+    sys.stderr.write(f'''
```

Comment:
can we have a flag to autoapprove this? I would image it would be helpful while developiong/testing.

Reply:
Due to our fun bash/Python hybrid, piping additional args is a bit difficult here. I split the difference and allowed skipping the prompt if the `CF_SKIP_WARN=1` environment variable has been set.

Reply:
please add a message when we're skipping.